hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java

```diff
@@ -571,23 +570,25 @@ public ServerName getRegionServerOfRegion(RegionInfo regionInfo) {
    * wants to iterate this exported list.  We need to synchronize on regions
    * since all access to this.servers is under a lock on this.regions.
    *
-   * @return A clone of current assignments.
+   * @return A clone of current open or opening assignments.
    */
   public Map<TableName, Map<ServerName, List<RegionInfo>>> getAssignmentsForBalancer(
       TableStateManager tableStateManager, List<ServerName> onlineServers) {
     final Map<TableName, Map<ServerName, List<RegionInfo>>> result = new HashMap<>();
     for (RegionStateNode node : regionsMap.values()) {
-      if (isTableDisabled(tableStateManager, node.getTable())) {
```

Comment:
nit: DisableTableProcedure first sets the table state to DISABLED and then force unassigns the regions in a loop, so we may want to retain this check to avoid weird races when "disable table" and the "chore" is running to totally avoid the table when it is marked DISABLED but regions are still a WIP.

Reply:
yeah, I think this makes sense. +1 to retaining this check.

Reply:
Yes, let's add it back. It is added in HBASE-23377, https://github.com/apache/hbase/pull/908

Reply:
Ok ,will do